### ISSUE TITLE
Webpush notifications for homescreen app aren’t working on iOS 16.4

### ISSUE THREAD

**Issue by Smexhy** [sct/overseerr#3398]
### Description

Hello, after the latest iOS 16.4 update, I tried to turn on webpush notifications for Overseerr since the ability to receive them was finally added to iOS natively but sadly, they aren’t working. 

This was the only webpush that I tried as a test on my iPhone and it worked just fine: https://devtails.xyz/@adam/how-to-setup-web-push-notifications-in-ios-safari , so I think there needs to be something changed for it to work with just specifically iOS/webkit/safari.

I am using HTTPS and also Jellyseerr, I think that’s worth mentioning at least, not that it should matter in this specific case.

### Version

Latest version

### Steps to Reproduce

1. Turn on webpush notifications in settings
2. Allow receiving notifications on iOS prompt
3. Try them on iOS 16.4 and any iPhone device (iOS 16.4 now officially supports webpush), after you try sending test notifications, it won’t work

### Platform

smartphone

### Device

iPhone 12 Pro

### Operating System

iOS 16.4

### Browser

Safari

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by aniqueta** [comment]
I have the same issue on iOS/iPadOS 16.4 (with Overseerr, not Jellyseerr) and after reinstalling the Home Screen icon, in case that made a difference. Web Push works for my setup with macOS Safari clients and other browsers. From quick read, the WebKit/Apple documentation[1][2][3] does not seem to differentiate macOS Safari and iOS/iPadOS Safari Web Push, so not sure why it doesn't "just work" and make life easier for our Overseerr devs.

[1] https://webkit.org/blog/13878/web-push-for-web-apps-on-ios-and-ipados/
[2] https://webkit.org/blog/12945/meet-web-push/
[3] https://developer.apple.com/documentation/usernotifications/sending_web_push_notifications_in_safari_and_other_browsers

**Comment by aniqueta** [comment]
Closing and opening the Home Screen app and the navigating around eventually triggered the notifications permissions request.  Web Push now works for me on iOS 16.4. I'm not sure what steps triggered the permissions prompt though.

**Comment by sct** [comment]
I have been using the 16.4 beta for a while without any issues. Push notifications have been working (and I believe the same for others). I am not sure why your devices are not receiving notifications. Still, we will be making some minor improvements to web push device visibility in the future so you can forcefully reset devices.

**Comment by sambartik** [comment]
Hello everyone,

I have taken a peak at this myself. First of all, I was having troubles getting the permission window to pop up. I tried multiple times readding the Overseerr to the home screen and despite that it did not work. So I cheated a little and requested the permissions myself from a console. (maybe having a separate button for that would work better on iphones?)

After that I stumbled upon another error. I couldn't send a test webpush notification from the admin settings. That was weird, but debugging hinted that it was because of a _Content Blocker_ installed in Safari. More specifically - the _Adguard Pro - Social_ list was making all the fuss.

Lastly, when the notification permission was granted and when I finally got the Test webpush notification button to work, I still haven't received a notification on an iPhone. But my Firefox browser on a laptop (registered prior the testing) received the push notification - suggesting that the notification(s) were sent out correctly from the server.

So I decided to peek into the Overseer's SQLite database and checked for users' registered push subscriptions. My Overseer's user ID had an additional entry there with the iPhone's subscription URL, which I verified to be the same as in the PushManager of the service worker on the iPhone. I suppose the keys were then the same as well.

This is where I stopped and hit a roadblock. My hypothesis is that Apple is being stricter with the notification validation on their part. I have a feeling it might be because my instance is not exposed directly to the internet. It has a private IP, behind a NAT and its HTTPS certificate is signed by a local CA.

Cheers!
